File: Data Structures & Algorithms/binary-tree-right-side-view/submission-2.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def rightSideView(self, root: Optional[TreeNode]) -> List[int]:
        rightview = []
        deck = deque()
        if root:
            deck.append(root)
        while deck:
            arr = []
            for i in range(len(deck)):
                x = deck.popleft()
                arr.append(x.val)
                if x.left:
                    deck.append(x.left)
                if x.right:
                    deck.append(x.right)
            rightview.append(arr[-1])
        return rightview

File: Data Structures & Algorithms/count-good-nodes-in-binary-tree/submission-2.js
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     constructor(val = 0, left = null, right = null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    /**
     * @param {TreeNode} root
     * @return {number}
     */
    foo(node, max_val) {
        const c3 = node.val >= max_val ? 1: 0;

        max_val = Math.max(max_val, node.val);
        
        const c1 = node.left ? this.foo(node.left, max_val) : 0;
        const c2 = node.right ? this.foo(node.right, max_val): 0;

        return c1 + c2 + c3;
    }

    goodNodes(root) {
        // root cannot be null as given that there is at least one node
        return this.foo(root, -Infinity);
    }
}
